src/components/modal/modal.jsx

```diff
@@ -0,0 +1,195 @@
+  @media (min-width: 1000px) {
+    display: none;
+  }
+`;
+const ModalTitle = styled.h5`
+  font-style: normal;
+  font-weight: 400;
+  font-size: 10px;
+  line-height: 120%;
+  text-align: center;
+  text-transform: uppercase;
+  font-feature-settings: "pnum" on, "lnum" on;
+  color: rgba(246, 246, 246, 0.7);
+  margin: 32px 0 16px;
+  @media (min-width: 1000px) {
+    margin: 84px 0 12px;
+  }
+`;
+const ModalNotificationBox = styled.div`
+  display: flex;
+  flex-direction: column;
+`;
+
+const ModalText = styled.p`
+  font-style: normal;
+  font-weight: 700;
+  font-size: 18px;
+  line-height: 120%;
+  text-align: center;
+  font-feature-settings: "pnum" on, "lnum" on;
+  color: #f6f6f6;
+  margin-bottom: 32px;
+`;
+const ModalFooter = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-self: flex-end;
+`;
+const ModalCloseButton = styled.div`
+  box-sizing: border-box;
+  margin: 0;
+  padding: 0;
+  position: absolute;
+  top: 26px;
+  right: 26px;
+  width: 12px;
+  height: 12px;
+  background-color: transparent;
+  background-image: url("${closeIcon}");
+  background-position: center;
+  background-size: cover;
+  border: none;
+  cursor: pointer;
+  transition: opacity 0.3s;
+  @media (max-width: 1000px) {
+    display: none;
+  }
+`;
+
+export const Modal = ({ isOpen, onClose, title, text, children }) => {
+  const y = useMotionValue(0);
+  const isLessThan1000 = useMediaQuery("(max-width:1000px)");
+  useEffect(() => {
+    if (!isOpen) {
+      onClose();
+    }
+  }, [isOpen]);
+
+  return (
+    <AnimatePresence>
+      {" "}
+      {isOpen && (
+        <ModalComponent
+          className="modal"
+          role="modal"
+          initial={{y: isLessThan1000 ? "100%" :"", opacity: 0 }}
+          animate={{ y: isOpen ? 0 : "100%", opacity: 1 }}
+          exit={{ y: isLessThan1000 ?"100%":"", opacity: 0 }}
+          transition={{
+            duration: 0.3,
+            type: "spring",
+            damping: 50,
+            stiffness: 200,
+          }}
+          drag="y"
+          dragElastic={0.1}
+          dragMomentum={false}
+          dragConstraints={{ top: 0, bottom: 0 }}
+          onDragEnd={(event, info) => {
+            if (info.offset.y > 300) {
+              onClose();
+            }
+          }}
+        >
+          <ModalWrapper>
+            <ModalCloseButton
+              type="button"
+              onClick={onClose}
+              aria-label="Close modal"
+              data-bs-dismiss="modal"
+            ></ModalCloseButton>
+            <ModalBody>
+              <ModalHeader />
+              <ModalTitle>{title}</ModalTitle>
+              <ModalNotificationBox>
+                {children}
+                <ModalText>{text}</ModalText>
+              </ModalNotificationBox>
+            </ModalBody>
+            <ModalFooter>{children}</ModalFooter>
+          </ModalWrapper>
+        </ModalComponent>
+      )}
+    </AnimatePresence>
+  );
+};
+
+Modal.Header = ModalHeader;
```

Comment:
For namespacing to be effective, wouldn't you need to use the namespace for the component (ex <Modal.Title> instead of <ModalTitle> )? @jahorwitz

Reply:
I'm not sure I understand what you mean here-- can you clarify the question?